bftengine/src/bftengine/SeqNumInfo.cpp

```diff
@@ -405,6 +618,127 @@ IncomingMsgsStorage& SeqNumInfo::ExFuncForCommitCollector::incomingMsgsStorage(v
   return r->getIncomingMsgsStorage();
 }
 
+///////////////////////////////////////////////////////////////////////////////
+// class SeqNumInfo::ExFuncForFastPathOptimisticCollector
+///////////////////////////////////////////////////////////////////////////////
+
+FullCommitProofMsg* SeqNumInfo::ExFuncForFastPathOptimisticCollector::createCombinedSignatureMsg(
+    void* context,
+    SeqNum seqNumber,
+    ViewNum viewNumber,
+    const char* const combinedSig,
+    uint16_t combinedSigLen,
+    const concordUtils::SpanContext& span_context) {
+  InternalReplicaApi* r = (InternalReplicaApi*)context;
+
+  return new FullCommitProofMsg(
```

Comment:
it would be good to check if this message is deleted by the caller.

Reply:
This method is called internally by CollectorOfTrhesholdSignatures and the caller becomes the owner of the message.